WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -81,12 +81,26 @@ def edit
   end
 
   def destroy
-    @registration = Registration.find(params[:id])
-    @registration.destroy
-    mailer = RegistrationsMailer.notify_registrant_of_deleted_registration(@registration)
-    mailer.deliver_now
-    flash[:success] = "Deleted registration and emailed #{mailer.to.join(" ")}"
-    redirect_to competition_edit_registrations_path(@registration.competition)
+    if !current_user
```

Comment:
This works, but the way we normally handle this is to add `:destroy` to the list of `before_action :authenticate_user!`. You can see we have that on line 2, and I believe it actually already includes this `destroy` method, so you can just get rid of this if statement entirely.

If we don't already have one, we should add a spec for someone who is not logged in trying to delete a registration.